public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/SimplifiedRouting.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useReducer } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, Stack, Text, useStyles2 } from '@grafana/ui';
+import { RuleFormValues } from 'app/features/alerting/unified/types/rule-form';
+import { createUrl } from 'app/features/alerting/unified/utils/url';
+
+import {
+  AlertManagerMetaData,
+  useGetAlertManagersMetadata,
+} from '../../notificaton-preview/useGetAlertManagersSourceNamesAndImage';
+
+import { ContactPointSelector } from './ContactPointSelector';
+
+export interface AMContactPoint {
+  alertManager: AlertManagerMetaData;
+  selectedContactPoint?: string;
+}
+
+export const selectContactPoint = createAction<{ receiver: string | undefined; alertManager: AlertManagerMetaData }>(
+  'simplifiedRouting/selectContactPoint'
+);
+
+export const receiversReducer = createReducer<AMContactPoint[]>([], (builder) => {
+  builder.addCase(selectContactPoint, (state, action) => {
+    const { receiver, alertManager } = action.payload;
+    const newContactPoint: AMContactPoint = { selectedContactPoint: receiver, alertManager };
+    const existingContactPoint = state.find((cp) => cp.alertManager.name === alertManager.name);
+
+    if (existingContactPoint) {
+      existingContactPoint.selectedContactPoint = receiver;
+    } else {
+      state.push(newContactPoint);
+    }
+  });
+});
+
+export function SimplifiedRouting() {
+  const { getValues, setValue } = useFormContext<RuleFormValues & { location?: string }>();
+  const styles = useStyles2(getStyles);
+  const contactPointsInAlert = getValues('contactPoints');
+
+  const alertManagerMetaData = useGetAlertManagersMetadata();
+
+  const alertManagerMetaDataPostable = alertManagerMetaData.filter((am) => am.postable);
+
+  // we merge the selected contact points with the alert manager meta data
+  const alertManagersWithSelectedContactPoints = alertManagerMetaDataPostable.map((am) => {
+    const selectedContactPoint = contactPointsInAlert?.find((cp) => cp.alertManager === am.name);
+    return { alertManager: am, selectedContactPoint: selectedContactPoint?.selectedContactPoint };
+  });
+
+  // use reducer to keep this alertManagersWithSelectedContactPoints in the state
+  const [alertManagersWithCPState, dispatch] = useReducer(receiversReducer, alertManagersWithSelectedContactPoints);
+
+  function getContactPointsForForm(alertManagersWithCP: AMContactPoint[]) {
+    return alertManagersWithCP.map((am) => {
+      return { alertManager: am.alertManager.name, selectedContactPoint: am.selectedContactPoint };
+    });
+  }
+
+  // whenever we update the receiversState we have to update the form too
+  useEffect(() => {
+    const contactPointsForForm = getContactPointsForForm(alertManagersWithCPState);
+    setValue('contactPoints', contactPointsForForm, { shouldValidate: false });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Having to require to silence the warning here is a code smell – why was this disabled?

Reply:
I forgot to remove it